### DISCUSSION TITLE
Error in TID

### DISCUSSION THREAD

**Discussion by marcelogcampos** [FeynCalc/feyncalc#231] [Q&A] [answered]
Hi Vlad,

when we try to perform the tensor decomposition (TID command) of our four dimensional amplitude, we came across with the following message:

"Error! TID has encountered a fatal problem and must abort the computation. The problem reads: "Your input contains a mixture of 4- and D-dimensional quantities. This is in general not allowed in dimensional regularization, unless you are using the Breitenlohner-Maison-t'Hooft-Veltman scheme.""

For the same amplitude, when we try to perform the ToPaVe command, the next message appears:

"Error! ToPaVe has encountered a fatal problem and must abort the computation. The problem reads: Not all 1-loop scalar integrals could be converted to PaVe functions. Please apply FDS to the input and try again."

We apply the FDS instruction but the problem persists.

Our amplitude integral is

~ⅈ gs^2 g^{Lor1 Lor2} g^{Lor10 Lor9} g^{Lor3 Lor4} g^{Lor5 Lor6} g^{Lor7 Lor8}
f^{ Glu7 Glu9 Glu11} f^{Glu8 Glu10 Glu11} T_{Col5 Col6}^{Glu8} T_{Col3 Col4}^{Glu7} T_{Col2 Col7}^{Glu10} T_{Col7 Col1}^{Glu9}
(g_{Lor4 Lor8} (k3 + k4 + p2 + q){Lor10} +
g{Lor10 Lor8} (k3 + k4 - 2p2 - 2q){Lor4} +
g{Lor10 Lor4} (-2k3 - 2k4 + p2 + q)Lor8)
(g{Lor6 Lor9} (-k1 - k2 - 2k3 - 2k4 + 2p2 + 2q){Lor2} +
g{Lor2 Lor9} (-k1 - k2 + k3 + k4 - p2 - q){Lor6} +
g{Lor2 Lor6} (2k1 + 2k2 + k3 + k4 - p2 - q){Lor9})
(φ ( k3 , md )) . (-ⅈ gs γ{Lor3}) . (φ ( -k4 , md ))
(φ ( k1 , mu )) . (-ⅈ gs γ_{Lor1}) . (φ ( -k2 , mu ))
(φ ( -p2 , mc )) . (-ⅈ gs γ_{Lor7} ) . (γ ·q + mc) .
(-ⅈ gs γ_{Lor5}) . (φ ( p1 , mc ))) /
(16 π^4 ( -k1 - k2 )^2 (-k3 - k4 )^2 (q^2 - mc^2) . (p2 + q )^2 . (-k3 - k4 + p2 + q )^2 . (-k1 - k2 - k3 - k4 + p2 + q )^2))) 

Could help us again?

Fraternally,
marcelo.

**Comment by vsht** [comment, marked as answer]
Hi Marcello,

from your expression I cannot really see the dimensions of the
quantities involved (you should post the StandardForm output,
not the TraditionalForm one), but judging from the error message
some of them are 4-dimensional, which is not correct in NDR (naive 
dimensional regularization). The quick fix would be to promote 
everything to D dimensions using ChangeDimension:

https://feyncalc.github.io/FeynCalcBookDev/ChangeDimension.html

Regarding ToPave, that command works only on scalar integrals while
in your case a tensor reduction is needed. So ToPaVe is not really
helpful at this stage.

Cheers,
Vlad

Am 17.09.23 um 00:53 schrieb marcelogcampos:

**Reply by marcelogcampos** [reply]
Hi Vlad,

Thank you for your help.
We changed the dimension amplitude to "D" and submitted the equation to TID command. It seems the calculation will take overnight...

Fraternally,
marcelo.

**Comment by vsht** [comment]
Hi Marcello,

you should have a look at Sec 4.1 in

https://arxiv.org/pdf/2001.04407.pdf

Most likely the full reduction is not necessary
or you are not specifying the relevant kinematic
constraints.

Cheers,
Vlad

Am 19.09.23 um 01:58 schrieb marcelogcampos:

**Reply by marcelogcampos** [reply]
Hi Vlad,

Thank you again for help.

We set the kinematic constraints. The processing time is still long. 
The amplitude carries around 10 different equations which have rank 1, 2 or 3. Each one of them has the numerator basically formed by 3 fermionic currents, where one of them has a fermionic propagator between the spinors.
We will try some symplifications and, maybe, try to do the tensor reduction at each one of these equations separatelly, once they have the same Feynman propagator.

Have a nice thursday.
Fraternally,
marcelo.

**Reply by vsht** [reply]
Can you post some sample example (with kinematics) that you are reducing with `TID`?

**Reply by marcelogcampos** [reply]
Hi Vlad,

One of the "full" (containing the spinorial structure and Dirac Gamma matrices) expressions is 

(\[CurlyPhi] (Subscript[k, 1], Subscript[m, u])) . (\[Gamma] \[CenterDot] (2q - Subscript[k, 1] - Subscript[k, 2] - 2 Subscript[k, 3] - 2 Subscript[k, 4] + 2 Subscript[p, 2])) . (\[CurlyPhi] (-Subscript[k, 2], Subscript[m, u])) 
(\[CurlyPhi] (Subscript[k, 3], Subscript[m, d])) . (\[Gamma]\[CenterDot]  (-2q + Subscript[k, 3] + Subscript[k, 4] - 2 Subscript[p, 2])) . (\[CurlyPhi](-Subscript[k, 4], Subscript[m, d])) 
(\[CurlyPhi] (-Subscript[p, 2], Subscript[m, c])) . \[Gamma]^\[Chi] . (\[Gamma]\[CenterDot]q + Subscript[m, c]) . \[Gamma]^\[Chi] . (\[CurlyPhi] (Subscript[p, 1], Subscript[m, c])) / 
((q^2 - Subsuperscript[m, c, 2]) . (q + Subscript[p, 2])^2 . (q - Subscript[k, 3] - Subscript[k, 4] + Subscript[p, 2])^2 . (q - Subscript[k, 1] - Subscript[k, 2] - Subscript[k, 3] - Subscript[k, 4] + Subscript[p, 2])^2  (-Subscript[k, 1]-Subscript[k, 2])^2  (-Subscript[k, 3]-Subscript[k, 4])^2)


This same expression free from the spinorial structure and from the Dirac Gamma matrices (with only the Feynman propagator and the rank 1 tensors q^\[Mu], q^\[Nu] and q^\[Rho]), which we are reducing with TID is

q^\[Mu] q^\[Nu] q^\[Rho]  / 
((q^2 - Subsuperscript[m, c, 2]) . (q + Subscript[p, 2])^2 . (q - Subscript[k, 3] - Subscript[k, 4] + Subscript[p, 2])^2 . (q - Subscript[k, 1] - Subscript[k, 2] - Subscript[k, 3] - Subscript[k, 4] + Subscript[p, 2])^2  (-Subscript[k, 1]-Subscript[k, 2])^2  (-Subscript[k, 3]-Subscript[k, 4])^2)  

where "q" is loop momentum and the kinematic constraints are

SPD[Subscript[p,1]]=SMP["m_c"]^2
SPD[Subscript[p,2]]=SMP["m_c"]^2
SPD[Subscript[k,1]]=SMP["m_u"]^2
SPD[Subscript[k,2]]=SMP["m_u"]^2
SPD[Subscript[k,3]]=SMP["m_d"]^2
SPD[Subscript[k,4]]=SMP["m_d"]^2

Fraternally,
marcelo.

**Reply by vsht** [reply]
Hi Marcelo,

could you please post a code that one can copy to a Mathematica notebook? The current output is not suitable for that, unfortunately. You can try to put that back into a notebook yourself to see that this is not a valid FeynCalc expression.

Cheers,
Vlad

**Reply by marcelogcampos** [reply]
Hi Vlad,

Let me try again...
Firstly, the "full" expression (with spinorial and Dirac matrices):

Spinor[Momentum[Subscript[k, 1], D], SMP["m_u"]] . GSD[2 q - Subscript[k, 1] - Subscript[k, 2] - 2 Subscript[k, 3] - 2 Subscript[k, 4] + 2 Subscript[p, 2]]. Spinor[Momentum[-Subscript[k, 2], D], SMP["m_u"]] Spinor[Momentum[Subscript[k, 3], D], SMP["m_d"]] . GSD[-2 q + Subscript[k, 3] + Subscript[k, 4] - 2 Subscript[p, 2]] . Spinor[Momentum[Subscript[-k, 4], D], SMP["m_d"]] Spinor[Momentum[Subscript[-p, 2], D], SMP["m_c"]] . GAD[\[Chi]] . (GSD[q] + SMP["m_c"]) . GAD[\[Chi] ]. Spinor[Momentum[Subscript[p, 1], D], SMP["m_c"]] FAD[{q, SMP["m_c"]}, q + Subscript[p, 2], q + Subscript[p, 2] - Subscript[k, 3] - Subscript[k, 4], q + Subscript[p, 2] - Subscript[k, 1] - Subscript[k, 2] - Subscript[k, 3] - Subscript[k, 4]]


and the "simplified" expression (free from spinorial structure and Dirac matrices):


FAD[{q, SMP["m_c"]}, q + Subscript[p, 2], q + Subscript[p, 2] - Subscript[k, 3] - Subscript[k, 4], q + Subscript[p, 2] - Subscript[k, 1] - Subscript[k, 2] - Subscript[k, 3] - Subscript[k, 4]] FVD[q, \[Mu]] FVD[q, \[Nu]] FVD[q, \[Xi]]


The kinematics:

SPD[Subscript[p, 1]] = SMP["m_c"]^2
SPD[Subscript[p, 2]] = SMP["m_c"]^2
SPD[Subscript[k, 1]] = SMP["m_u"]^2
SPD[Subscript[k, 2]] = SMP["m_u"]^2
SPD[Subscript[k, 3]] = SMP["m_d"]^2
SPD[Subscript[k, 4]] = SMP["m_d"]^2

I guess these will work. If they dont, please let me know and I´ll send you a different version.

Fraternally,
marcelo.

**Reply by vsht** [reply]
Hi Marcelo,

the reduction to coefficient functions doesn't require much time
```
res = TID[amp, q, FCVerbose -> 1, UsePaVeBasis -> True];
```
But reducing those D-functions to scalar integrals will generate huge
expressions
```
pave = Cases2[res // ToPaVe2, PaVe]

pave[[1]] // PaVeReduce
```
To me it doesn't make much sense, because you have too many free parameters
(p1.p2, k1.k3 etc.) to have a nice analytic result and the numerics with scalar integrals
and that many kinematic invariants is likely to be extremely unstable.

So I'd rather take `res` and pass that to LoopTools or Collier without trying to simplify
things further.

Cheers,
Vlad

**Reply by marcelogcampos** [reply]
Hi Vlad,

Thank you for your suggestions and for the time dedicated to us. We appreciate it. 

We were performing the TID command for that expression without setting "UsePaVeBasis->True". That´s why it was taking so long to evaluate it.

You mentioned about LoopTools and Collier. We have no expirience with them. What are the their advantages? Do the Fermat and PackageX work in this case, as well? And, what about, for example, FormCalc or CalcHEP? Do you think these packages are good options for us?

Fraternally,
marcelo.

**Comment by vsht** [comment]
Hi Marcelo,

>> You mentioned about LoopTools and Collider. We have no expirience with 
>> them. What are the their advantages? And, what about, for example, 
>> FormCalc or CalcHEP? Do you think these packages are good options for us?
>> 


those are different packages used in loop calculations that serve
different purposes. There are lots of tutorials and reviews you
can find on that and since the software is also publicly available,
you should have no difficulties to try them out.

I'm sorry for not giving you the answer you were expecting, but
advising people on calculating Feynman diagrams with computer tools
is not really something I can find time for, given the harsh reality
in academia. So I prefer to leave it at "How to do the step XYZ with 
FeynCalc".

I can only say that much: If you're really starting from scratch,
FormCalc might be a better fit for you, since it requires significantly
less user input and will get you faster to the plots/numerics you
want to produce. FeynCalc is better when you precisely understand
what you are doing and need more flexibility but it also requires a lot
of extra work.

Cheers,
Vlad









Am 28.09.23 um 22:50 schrieb marcelogcampos:

**Reply by marcelogcampos** [reply]
Hi Vlad,

We really do know how time is precious in academia, so we can imagine how hard is to you to dedicate time and attention to answer our questions. It is a great benefit to us exchange these messages with you, one of the FeynCalc developers.

We want to thank you for all your help. We appreciate it.

Fraternally,
marcelo.
